cli/cli/helpers/grafloki/kubernetes_grafloki.go

```diff
@@ -0,0 +1,477 @@
+	existsGrafana := false
+	var lokiHost string
+	lokiDeployment, err := k8sManager.GetDeployment(ctx, graflokiNamespace, lokiDeploymentName)
+	if err == nil && lokiDeployment != nil {
+		existsLoki = true
+		lokiHost = getLokiHost(lokiServiceName, graflokiNamespace, lokiNodePort)
+	}
+
+	grafanaDeployment, err := k8sManager.GetDeployment(ctx, graflokiNamespace, grafanaDeploymentName)
+	if err == nil && grafanaDeployment != nil {
+		existsGrafana = true
+	}
+
+	return existsLoki && existsGrafana, lokiHost
+}
+
+func StopGrafLokiInKubernetes(ctx context.Context) error {
+	k8sManager, err := getKubernetesManager()
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred getting Kubernetes Manager.")
+	}
+	ns, err := k8sManager.GetNamespace(ctx, graflokiNamespace)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred getting graflokiNamespace '%v'.", graflokiNamespace)
+	}
+	err = k8sManager.RemoveNamespace(ctx, ns)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred removing graflokiNamespace '%v'.", graflokiNamespace)
+	}
+	err = waitForNamespaceRemoval(ctx, graflokiNamespace, k8sManager)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred while waiting for graflokiNamespace '%v' removal.", graflokiNamespace)
+	}
+	return nil
+}
+
+func getKubernetesManager() (*kubernetes_manager.KubernetesManager, error) {
+	kubernetesConfig, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		clientcmd.NewDefaultClientConfigLoadingRules(), nil,
+	).ClientConfig()
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred creating Kubernetes configuration")
+	}
+	clientSet, err := kubernetes.NewForConfig(kubernetesConfig)
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "Expected to be able to create Kubernetes client set using Kubernetes config '%+v', instead a non nil error was returned", kubernetesConfig)
+	}
+	k8sManager := kubernetes_manager.NewKubernetesManager(clientSet, kubernetesConfig, "")
+	return k8sManager, nil
+}
+
+func waitForNamespaceRemoval(
+	ctx context.Context,
+	namespace string,
+	kubernetesManager *kubernetes_manager.KubernetesManager) error {
+	var (
+		maxTriesToWaitForNamespaceRemoval       uint = 30
+		timeToWaitBetweenNamespaceRemovalChecks      = 1 * time.Second
+	)
+
+	for i := uint(0); i < maxTriesToWaitForNamespaceRemoval; i++ {
+		if _, err := kubernetesManager.GetNamespace(ctx, namespace); err != nil {
+			// if err was returned, graflokiNamespace doesn't exist, or it's been marked for deleted
+			return nil
+		}
+
+		// Tiny optimization to not sleep if we're not going to run the loop again
+		if i < maxTriesToWaitForNamespaceRemoval-1 {
+			time.Sleep(timeToWaitBetweenNamespaceRemovalChecks)
+		}
+	}
+
+	return stacktrace.NewError("Attempted to wait for namespace '%v' removal or to be marked for deletion '%v' times but '%v' was not removed.", namespace, namespace, maxTriesToWaitForNamespaceRemoval)
```

Comment:
There's a bug in the ordering of the vars printed. This would print:
```
Attempted to wait for namespace 'graflokiNamespace' removal or to be marked for deletion 'graflokiNamespace' times but '30' was not removed.
```